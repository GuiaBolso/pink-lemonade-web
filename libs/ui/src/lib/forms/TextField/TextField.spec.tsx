import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { TextField } from './index';

describe('TextField Component', () => {
  const makeSetup = () => {
    const element = (
      <TextField label="Lorem Ipsum" helperText="Lorem ipsum dolor sit amet." />
    );
    const utils = render(element);
    const input = utils.container.querySelector('input');
    const label = utils.container.querySelector('label.MuiFormLabel-root');
    const helperText = utils.container.querySelector(
      'p.MuiFormHelperText-root',
    );

    return {
      element,
      input,
      label,
      helperText,
      ...utils,
    };
  };

  const setup = makeSetup();

  it('It should be same value in front of the input', () => {
    const { input } = setup;
    fireEvent.change(input, { target: { value: 'Lorem ipsum dolor' } });
    expect(input.value).toBe('Lorem ipsum dolor');
  });

  it('It should be correct label', () => {
    const { label } = setup;
    expect(label.textContent).toBe('Lorem Ipsum');
  });

  it('It should be correct helper text', () => {
    const { helperText } = setup;
    expect(helperText.textContent).toBe('Lorem ipsum dolor sit amet.');
  });

  it('renders correctly', () => {
    const { element } = setup;
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
