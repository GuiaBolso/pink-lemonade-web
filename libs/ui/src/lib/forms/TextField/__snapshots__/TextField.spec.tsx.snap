// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-90vjmg"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
  >
    Lorem Ipsum
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-7"
      >
        <span>
          Lorem Ipsum
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-contained"
  >
    Lorem ipsum dolor sit amet.
  </p>
</div>
`;
